warehouse/packaging/tasks.py

```diff
@@ -357,90 +361,74 @@ def sync_bigquery_release_files(request):
     # Multiple table names can be specified by separating them with whitespace
     table_names = release_files_table.split()
 
-    for table_name in table_names:
-        table_schema = bq.get_table(table_name).schema
-
-        # Using the schema to populate the data allows us to automatically
-        # set the values to their respective fields rather than assigning
-        # values individually
-        def populate_data_using_schema(file):
-            release = file.release
-            project = release.project
-
-            row_data = dict()
-            for sch in table_schema:
-                # The order of data extraction below is determined based on the
-                # classes that are most recently updated
-                if hasattr(file, sch.name):
-                    field_data = getattr(file, sch.name)
-                elif hasattr(release, sch.name) and sch.name == "description":
-                    field_data = getattr(release, sch.name).raw
-                elif sch.name == "description_content_type":
-                    field_data = getattr(release, "description").content_type
-                elif hasattr(release, sch.name):
-                    field_data = getattr(release, sch.name)
-                elif hasattr(project, sch.name):
-                    field_data = getattr(project, sch.name)
-                else:
-                    field_data = None
-
-                if isinstance(field_data, datetime.datetime):
-                    field_data = field_data.isoformat()
-
-                # Replace all empty objects to None will ensure
-                # proper checks if a field is nullable or not
-                if not isinstance(field_data, bool) and not field_data:
-                    field_data = None
-
-                if field_data is None and sch.mode == "REPEATED":
-                    row_data[sch.name] = []
-                elif field_data and sch.mode == "REPEATED":
-                    # Currently, some of the metadata fields such as
-                    # the 'platform' tag are incorrectly classified as a
-                    # str instead of a list, hence, this workaround to comply
-                    # with PEP 345 and the Core Metadata specifications.
-                    # This extra check can be removed once
-                    # https://github.com/pypi/warehouse/issues/8257 is fixed
-                    if isinstance(field_data, str):
-                        row_data[sch.name] = [field_data]
+    missing_files = (
+        request.db.query(MissingDatasetFile)
+        .filter(MissingDatasetFile.processed.is_(None))
+        .limit(10)
+    )
+    for missing_file in missing_files:
+        missing_file.processed = False
+
+    # Commit these changes so another task doesn't pick up these files
+    request.db.commit()
```

Comment:
I think this is necessary to prevent other tasks from picking up these rows, but I'm not sure we can do this here?

Reply:
Hmmm, yeah trying to do this in a task results in:

`AssertionError: Transaction must be committed using the transaction manager`

Reply:
You can:

```python
    # Commit these changes so another task doesn't pick up these files
    request.tm.commit()
    request.tm.begin()
```